src/graphics/programlib/programlib_phong.js

```diff
@@ -192,7 +229,18 @@ pc.programlib.phong = {
         //////////////////////////////
         // GENERATE FRAGMENT SHADER //
         //////////////////////////////
+        var fshader;
         code = getSnippet(device, 'fs_precision');
+
+        if (options.customPS) {

```

Comment:
How about options.pixelShader instead of customPS? Basically I think we should not have words like 'custom', 'override' etc